### PR TITLE
Trengo Components

### PR DESCRIPTION
**Pull request by feyzullah**
### Actions
* Create a Contact(Find or create contact)
* Log a Voice Call
* Send a Team Chat Message
* Send a Message
* Send a WhatsApp Message Template
* Find a Contact
### Sources
* New Inbound Message
* Ticket Label Added
* New Internal Note
* New Outbound Message
* Phone Call Ended
* Phone Call Missed
* Phone Call Started

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/trengo/actions/log-a-voice-call/log-a-voice-call.mjs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      propDefinition: [
+        app,
+        "recordingUrl",
+      ],
+    },
+  },
+  async run ({ $ }) {
+    const resp = await this.app.logVoiceCall({
+      $,
+      channelId: this.channelId,
```
Comment (jcortes) [inline components/trengo/actions/log-a-voice-call/log-a-voice-call.mjs:59]: Seems like you don't need this property here
```suggestion
```

File: components/trengo/actions/send-a-team-chat-message/send-a-team-chat-message.mjs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  async run ({ $ }) {
+    if (!this.threadId && !this.toUserId) {
+      throw new ConfigurationError("Either `Thread ID` or `To User ID` should be set!");
+    }
+    if (!this.body && !this.attachmentIds) {
+      throw new ConfigurationError("Either `Body` or `Attachement IDs` should be set!");
+    }
+    const resp = await this.app.sendTeamChatMessage({
+      $,
+      channelId: this.channelId,
```
Comment (jcortes) [inline components/trengo/actions/send-a-team-chat-message/send-a-team-chat-message.mjs:52]: Seems like you don't need this property here
```suggestion
```

**Review by jcortes — COMMENTED** [review]
Hi @feyzullah I've just added two minor request changes and you are done! Great work thanks!

**Review by jcortes — COMMENTED** [review]
HI @feyzullah LGTM! Ready for QA!

**Review by jcortes — APPROVED** [review]
Approved!

**Pull request merged**